src/reconcile.rs

```diff
@@ -271,9 +417,68 @@ fn pod_owned_by(pod: &Pod, owner_uid: &str) -> bool {
     matches!(controller, Some(OwnerReference { uid, .. }) if uid == owner_uid)
 }
 
+/// Helper method to make sure that any LabelSelector we use only matches our own "special" nodes.
+/// At the moment this label is "type" with the value "krustlet" and we'll use match_labels.
+///
+/// WARN: Should a label "type" already be used this will be overridden!
+/// If this is really needed add a match…expression
+///
+/// We will not however change the original LabelSelector, a new one will be returned.
+fn add_stackable_selector(selector: &LabelSelector) -> LabelSelector {
+    let mut selector = selector.clone();
+    selector
+        .match_labels
+        .get_or_insert_with(BTreeMap::new)
+        .insert("type".to_string(), "krustlet".to_string());
```

Comment:
@soenkeliebau Is this the correct label to target? Can you tell me (again) which would be best?

Reply:
https://github.com/deislabs/krustlet/blob/2c01e500b9cd906837b114ca00b401fefb21f363/crates/kubelet/src/node/mod.rs#L445

Reply:
Yeah but won't that hit any Krustlet? (On the off-chance that someone has another krustlet with WASM running for example).

Shouldn't we use something stackable specific?

Reply:
That is true, but would require a change in the agent at the same time. I think I'd prefer to merge this as is and make that change at a later time (doesn't have to be much later).

Reply:
Come to think of it, we could have label creation shared between the agent and this to ensure it remains the consistent. I am hesitant to make the agent depend on this framework solely for that reason though.. then again, won't hurt much after the compiler did it's job optimizing stuff out I think ..

Reply:
So we don't have any identifying label that we can use? Don't we have one with "stackable-linux"?

Reply:
This is what a node currently has:

```
Labels:             beta.kubernetes.io/arch=stackable-linux
                    beta.kubernetes.io/os=stackable-linux
                    kubernetes.io/arch=stackable-linux
                    kubernetes.io/hostname=soenke-laptop
                    kubernetes.io/os=stackable-linux
                    type=krustlet
Annotations:        node.alpha.kubernetes.io/ttl: 0
                    volumes.kubernetes.io/controller-managed-attach-detach: true
Taints:             kubernetes.io/arch=stackable-linux:NoExecute
                    kubernetes.io/arch=stackable-linux:NoSchedule
                    node.kubernetes.io/unreachable:NoSchedule
```